src/EditorFeatures/CSharpTest/ForeachToFor/ForeachToForTests.cs

```diff
@@ -0,0 +1,1177 @@
+        }
+    }
+}
+
+class Explicit : IReadOnlyList<int>
+{
+    int IReadOnlyList<int>.this[int index] => throw new NotImplementedException();
+    int IReadOnlyCollection<int>.Count => throw new NotImplementedException();
+    IEnumerator<int> IEnumerable<int>.GetEnumerator() => throw new NotImplementedException();
+    IEnumerator IEnumerable.GetEnumerator() => throw new NotImplementedException();
+}
+";
+
+            var expected = @"
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+class Test
+{
+    void Method()
+    {
+        var list = new Explicit();
+        var {|Rename:list1|} = (IReadOnlyList<int>)(list);
+        for (var {|Rename:i|} = 0; i < list1.Count; i++)
+        {
+            var a = list1[i];
+            Console.WriteLine(a);
+        }
+    }
+}
+
+class Explicit : IReadOnlyList<int>
+{
+    int IReadOnlyList<int>.this[int index] => throw new NotImplementedException();
+    int IReadOnlyCollection<int>.Count => throw new NotImplementedException();
+    IEnumerator<int> IEnumerable<int>.GetEnumerator() => throw new NotImplementedException();
+    IEnumerator IEnumerable.GetEnumerator() => throw new NotImplementedException();
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task DoubleExplicitInterface()
+        {
+            var text = @"
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+class Test
+{
+    void Method()
+    {
+        var list = new Explicit();
+        foreach [||] (var a in list)
+        {
+            Console.WriteLine(a);
+        }
+    }
+}
+
+class Explicit : IReadOnlyList<int>, IReadOnlyList<string>
+{
+    int IReadOnlyList<int>.this[int index] => throw new NotImplementedException();
+    int IReadOnlyCollection<int>.Count => throw new NotImplementedException();
+    IEnumerator<int> IEnumerable<int>.GetEnumerator() => throw new NotImplementedException();
+    IEnumerator IEnumerable.GetEnumerator() => throw new NotImplementedException();
+
+    string IReadOnlyList<string>.this[int index] => throw new NotImplementedException();
+    int IReadOnlyCollection<string>.Count => throw new NotImplementedException();
+    IEnumerator<string> IEnumerable<string>.GetEnumerator() => throw new NotImplementedException();
+}
+";
+            await TestMissingInRegularAndScriptAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task DoubleExplicitInterfaceWithExplicitType()
+        {
+            var text = @"
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+class Test
+{
+    void Method()
+    {
+        var list = new Explicit();
+        foreach [||] (int a in list)
+        {
+            Console.WriteLine(a);
+        }
+    }
+}
+
+class Explicit : IReadOnlyList<int>, IReadOnlyList<string>
```

Comment:
please extract all this duplication into a helper string you can insert into these tests.

Reply:
this is part of test. without this, how user know what it is testing?